### PR TITLE
⭐️New: Add vue/no-static-inline-styles rule

### LINKED ISSUE #826: Rule proposal: vue/no-inline-styles
**Please describe what the rule should do:**

This rule detect inline styles in components.
It's (subjectively) a good practice to separate styles from the view layout, when possible. This rule detects static inline style objects when they contain literal values. `v-bind:style` and `:style` are acceptable because it's sometimes necessary to set styles based on data, computed property or methods result.

**What category should the rule belong to?**

- [x] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

The following pattern is considered a warning:
```vue
<span style="background-color: #FFF">Hello {{ name }}</span>;
```

The following patterns are not considered a warning::
```vue
<span :style="background-color: color }">Hello {{ name }}</span>;
<span v-bind:style="background-color: color }">Hello {{ name }}</span>;
```

**Additional context**
The rule enforce team members to use SFC `<style scope>` instead of inline style attributes.

### PR DESCRIPTION
**Pull request by ota-meshi**
This PR adds the `vue/no-static-inline-styles` rule.
`vue/no-static-inline-styles` rule reports the static inline style given to the element.

We recommend moving the reported inline styles to the `<style>` element.

---

Close #826

### CODE REVIEW

**Comment by cde06** [comment]
Thank you so much for this PR!! 
I hope it will be release soon.

File: lib/rules/no-static-inline-styles.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      "VAttribute[directive=false][key.name='style']" (node) {
+        context.report({
+          node,
+          messageId: 'forbiddenStyleAttr'
+        })
+      }
+    }
+    if (!context.options[0] || !context.options[0].allowBinding) {
+      visitor[
+        "VAttribute[directive=true][key.name.name='bind'][key.argument.name='style']"
```
Comment (hirokiosame) [inline lib/rules/no-static-inline-styles.js]: Should this be `[key.name='bind'][key.argument='style']` ?

Comment (mysticatea) [reply lib/rules/no-static-inline-styles.js]: After #807, AST has been changed a bit to support dynamic argument.

Comment (hirokiosame) [reply lib/rules/no-static-inline-styles.js]: Ah of course. Thank you for the explanation.

**Comment by cde06** [comment]
Hello,
Can we release this fix or are we waiting something else?
Thanks

**Pull request merged**